karapace/instrumentation/prometheus.py

```diff
@@ -0,0 +1,100 @@
+"""
+karapace - prometheus instrumentation
+
+Copyright (c) 2024 Aiven Ltd
+See LICENSE for details
+"""
+# mypy: disable-error-code="call-overload"
+
+from __future__ import annotations
+
+from aiohttp.web import middleware, Request, Response
+from karapace.rapu import RestApp
+from prometheus_client import CollectorRegistry, Counter, Gauge, generate_latest, Histogram
+from typing import Awaitable, Callable, ClassVar
+
+import logging
+import time
+
+LOG = logging.getLogger(__name__)
+
+
+class PrometheusInstrumentation:
+    METRICS_ENDPOINT_PATH: ClassVar[str] = "/metrics"
+    START_TIME_REQUEST_KEY: ClassVar[str] = "start_time"
```

Comment:
just wondering, why are you setting those to be variant?

Reply:
what do you mean variant?

Reply:
`ClassVar`

Reply:
ah sorry, missread, this is `ClassVar` to mark it as variant you need to mark it as `TypeVar`

Reply:
its to indicate that all the instances share the same variable, make sense

Reply:
not sure if you can combine it with `Final` but I would prefer it, does it make sense to change the content? I would suggest to change it into `ClassVar[Final[str]]` or `Final[str]` if the first isn't allowed

Reply:
I remember I ran into an issue with mypy complaining about combining both `ClassVar` and `Final`, ideally as this would not have an instance, except for the class instance itself, one would have to directly change the class variable and this we can disallow if seen in a review.
If i would change at all, i'd change to `Final[T]`, wdyt?

Reply:
yeah, I would vote for changing into `Final`, we should understand why its complaining and probably there is a good reason 😄